Assets/_BForBoss/_Weapons/Scripts/Equipment/EquipmentBehaviour.cs

```diff
@@ -2,84 +2,63 @@
 using Perigon.Utility;
 using PerigonGames;
 using UnityEngine;
-using UnityEngine.InputSystem;
 
 namespace Perigon.Weapons
 {
     [RequireComponent(typeof(BulletSpawner))]
     [RequireComponent(typeof(WallHitVFXSpawner))]
     public partial class EquipmentBehaviour : MonoBehaviour
     {
-        [SerializeField] private InputActionAsset _inputActions;
         [SerializeField] private WeaponBehaviour[] _weaponBehaviours = null;
         private BulletSpawner _bulletSpawner;
         private WallHitVFXSpawner _wallHitVFXSpawner;
         private MeleeWeaponBehaviour _meleeBehaviour = null;
         private IWeaponAnimationProvider _weaponAnimationProvider;
+        private PGInputSystem _inputSystem;
         private Weapon[] _weapons = null;
-        private int _currentWeaponIndex = 0;
 
-        private InputAction _reloadInputAction = null;
-        private InputAction _fireInputAction = null;
-        private InputAction _swapWeaponInputAction = null;
-        private InputAction _meleeWeaponInputAction = null;
-        private bool _isMouseScrollEnabled = true;
+        private int _currentWeaponIndex = 0;
 
         private Transform _playerPivotTransform;
         
-        public void Initialize(Transform playerPivotTransform, IWeaponAnimationProvider weaponAnimationProvider)
+        public void Initialize(Transform playerPivotTransform, PGInputSystem inputSystem, IWeaponAnimationProvider weaponAnimationProvider)
```

Comment:
Refactored the  `EquipmentBehaviour`
- Previously - Swapping weapons is complicated, (2 animations needed, call different one depending if you scroll up or down.
- Currently - ONE animation only, and it will swap to correct weapon.

New:
Press keyboard 1, 2, 3 to swap weapons.